### PR TITLE
First draft of the offchain component tutorial

### PR DESCRIPTION
**Pull request by dubbelosix**
# Description
Adding TUTORIAL.md which explains the process of creating offchain processing and storage for the `sov-nft-module`
The tutorial covers
* Basic setup (installing postgres, creating the necessary tables)
* Creating offchain.rs and creating the functions encoding the offchain functionality to update the three tables - collections, nfts, top_owners
* Explains how to use the `offchain` macro
* Adding the offchain feature to the necessary crates to ensure it can be used for `demo-rollup`
* Using the nft-utils script to create some collections and nfts to see the results of the offchain functions

### CODE REVIEW

**Comment by preston-evans98** [comment]
This looks really nice. Great work!

**Review by neysofu — APPROVED** [review]

File: module-system/module-implementations/sov-nft-module/TUTORIAL.md [hunk 1]
```diff
@@ -0,0 +1,562 @@
+# Offchain Computation Tutorial
+
+## Introduction
+This tutorial serves to explain the design and process of augmenting Sovereign SDK modules with offchain computation and storage capabilities
```
Comment (cemozerr) [inline module-system/module-implementations/sov-nft-module/TUTORIAL.md]: How about something simpler such as:

```suggestion
This guide demonstrates how to add off-chain compute and storage capabilities to Sovereign SDK modules by walking you through a simple indexer implementation.
```

File: module-system/module-implementations/sov-nft-module/TUTORIAL.md [hunk 2]
```diff
@@ -0,0 +1,562 @@
...
+   owner              TEXT   NOT NULL,
+   collection_address TEXT   NOT NULL,
+   count              BIGINT NOT NULL,
+   PRIMARY KEY (owner, collection_address)
+);
+```
+* This table monitors each user in the system and the number of NFTs they own within a specific collection.
+* With this data, you can deduce:
+  * Who is the predominant owner for a particular collection, and how many NFTs they possess?
+  * What does the ownership distribution look like for each collection?
```
Comment (cemozerr) [inline module-system/module-implementations/sov-nft-module/TUTORIAL.md]:
```suggestion
  * How does the ownership distribution look like for each collection?
```

File: module-system/module-implementations/sov-nft-module/TUTORIAL.md [hunk 3]
```diff
@@ -0,0 +1,562 @@
...
+* For convenience, [init_db.sql](src/init_db.sql) provides a script that initializes the three aforementioned tables and the necessary indexes to enhance query speed. It's worth noting that the script removes any pre-existing tables before recreating them.
+
+### 3. Create the Offchain Functions
+* **Initialization**: With the tables set up, it's time to add offchain functionality to the [sov-nft-module](../sov-nft-module).
+* **Using the `#[offchain]` proc macro**: This is available in the [sov-modules-macros](../../sov-modules-macros/src/lib.rs) crate. Functions marked with this macro will only execute by the rollup when the `offchain` feature flag is active.
+  - If the `offchain` feature is enabled: the function is present as defined.
+  - If the `offchain` feature is absent: the function body is replaced with an empty definition.
+* The offchain macro is used to annotate functions that should only be executed by the rollup when the `offchain` feature flag is passed.
+* The macro produces one of two functions depending on the presence flag.
```
Comment (cemozerr) [inline module-system/module-implementations/sov-nft-module/TUTORIAL.md]: I believe we're repeating ourselves here. Line 87 and 88 looks similar.

**Review by cemozerr — APPROVED** [review]
Looks great! Left a few comments.

**Comment by cemozerr** [comment]
Also, can we add a reading time widget as in the Demo rollup?

**Pull request merged**